### ISSUE TITLE
Scritable CRUD?

### ISSUE THREAD

**Issue by alex-phillips** [budgetzero/budgetzero#27]
I know this is 'offline-first' with pouch DB, but is there a way to write scripts against this that can essentially act as an "API"? Would be great if I could write a script that could not only import transactions, but assigned them to categories, or if those categories don't exist yet, create them programmatically.

**Comment by stefanjf** [comment]
Yes this is possible. The client uses PouchDB and I don't know of an exposed API. But it's possible to sync it to a CouchDB server which has an API you could hit. 

Give me a few days to come out with the CouchDB server instructions. I'll also give you the database schema and some examples.

**Comment by alex-phillips** [comment]
Any update on this? With YNAB's latest price change, something like this could make migrating data from YNAB into BudgetZero fairly easy.

**Comment by jhonderson** [comment]
A good approach to achieve this is layering this project into a presentation layer and application layer. The application layer would expose all the functionality related to the budget, and it can be exposed as a reusable npm package. In that way it can be used not only by this website, but also to create an http api.

If we create an http api with all the logic to store the budget information in CouchDB we would be duplicating the logic in the website, and that would be hard to maintain.

I wonder what are the future plans for this budgetzero project, I don’t see any recent activity.
